tools/sz_repo_cli/lib/src/commands/src/test_command.dart

```diff
@@ -25,70 +32,55 @@ class TestCommand extends Command {
-
-  Duration packageTimeout;
-
   @override
-  Future<void> run() async {
-    packageTimeout = argResults.packageTimeoutDuration;
+  Future<Null> run() async {
     isVerbose = argResults['verbose'] ?? false;
 
-    await _testFlutterApp(_repo.sharezoneFlutterApp);
-    await _testPackages(_repo.dartLibraries);
-
-    print('All tests are passing!');
-  }
-
-  Future _testFlutterApp(Package flutterApp) async {
-    print('Starting flutter app tests...');
-
-    try {
-      await flutterApp.runTests().timeout(packageTimeout);
-    } catch (e) {
-      // Weil der Error ansonsten nicht ausgeprintet wird
-      print('$e');
-      print('Not all flutter app tests passed!');
-      throw ToolExit(1);
-    }
-
-    print('All flutter app tests passed!');
-  }
-
-  Future<void> _testPackages(DartLibraries dartLibraries) async {
-    final failingPackages = <String>[];
-    await for (final package in dartLibraries
-        .streamPackages()
-        .where((package) => package.hasTestDirectory)) {
-      final packageName = package.name;
-
-      print(
-          'RUNNING $packageName ${[package.type.toReadableString()]} tests...');
-
-      try {
-        await package.runTests().timeout(packageTimeout);
-        print('COMPLETED $packageName tests');
-      } catch (e) {
-        print('FAILURE: $e');
-        failingPackages.add(packageName);
-      }
-    }
-
-    print('\n\n');
-    if (failingPackages.isNotEmpty) {
-      print('Tests for the following packages are failing (see above):');
-      failingPackages.forEach((String package) {
-        print(' * $package');
-      });
-      throw ToolExit(1);
+    final _max = argResults[maxConcurrentPackagesOptionName];
+    final maxNumberOfPackagesBeingProcessedConcurrently = _max != null
+        ? int.tryParse(argResults[maxConcurrentPackagesOptionName])
+        // null wird nachher als "keine Begrenzung" gehandhabt.
+        : null;
+
+    final taskRunner = ConcurrentPackageTaskRunner(
+      getCurrentDateTime: () => DateTime.now(),
+    );
+
+    final res = taskRunner
+        .runTaskForPackages(
+          packageStream: repo
+              .streamPackages()
+              .where((package) => package.hasTestDirectory),
+          runTask: (package) => package.runTests(),
+          maxNumberOfPackagesBeingProcessedConcurrently:
+              maxNumberOfPackagesBeingProcessedConcurrently,
+          perPackageTaskTimeout: argResults.packageTimeoutDuration,
+        )
+        .asBroadcastStream();
+
+    final presenter = PackageTasksStatusPresenter();
+    presenter.continuouslyPrintTaskStatusUpdatesToConsole(res);
+
+    final failures = await res.allFailures;
+
+    if (failures.isNotEmpty) {
+      print('There were failures. See above for more information.');
+      await presenter.printFailedTasksSummary(failures);
+      exit(1);
+    } else {
+      print('All packages tested successfully!');
+      exit(0);
     }
-    print('All Dart package tests passed!');
   }
-}
+}
```

Comment:
```suggestion
}

```

Nit: Add newline at end of file.

Isn't VS Code adding a newline at the of file when you are pressing auto-format?

Reply:
No locally it adds a new line at the end when I save and the file is auto-formatted.

Reply:
For me locally there is a new line at the end - maybe it has something to do with git?

Reply:
I added a new line via GitHub but now there are two for me locally 🤔